warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java

```diff
@@ -4197,6 +4198,207 @@ public static GeoTimeSerie sortedMerge(GeoTimeSerie base, GeoTimeSerie gts) {
     return merged;
   }
 
+  /**
+   * Merge sorted GTS or encoders
+   */
+  public static GTSEncoder sortedMerge(List<Object> params, boolean reversed) throws WarpScriptException {
+
+    final class GTSAndIndex {
+      private GeoTimeSerie gts;
+      private int idx;
+    }
+
+    List<GeoTimeSerie> series = new ArrayList<GeoTimeSerie>();
+    List<GTSEncoder> encoders = new ArrayList<GTSEncoder>();
+
+    for (int i = 0; i < params.size(); i++) {
+      if (params.get(i) instanceof GeoTimeSerie) {
+        if (!GTSHelper.isSorted((GeoTimeSerie) params.get(i))) {
+          throw new WarpScriptException("GTS " + GTSHelper.buildSelector((GeoTimeSerie) params.get(i), false) + " is not sorted.");
+        }
+        if (GTSHelper.isReversed((GeoTimeSerie) params.get(i)) != reversed) {
+          throw  new WarpScriptException("GTS " + GTSHelper.buildSelector((GeoTimeSerie) params.get(i), false) + " is not sorted in the expected order.");
+        }
+        series.add((GeoTimeSerie) params.get(i));
+      } else if (params.get(i) instanceof GTSEncoder) {
+        encoders.add((GTSEncoder) params.get(i));
+      } else if (params.get(i) instanceof List) {
+        for (Object o: (List) params.get(i)) {
+          if (o instanceof GeoTimeSerie) {
+            if (!GTSHelper.isSorted((GeoTimeSerie) o)) {
+              throw new WarpScriptException("GTS " + GTSHelper.buildSelector((GeoTimeSerie) o, false) + " is not sorted.");
+            }
```

Comment:
Here, need to manage empty GTS (created by NEWGTS) (not sorted, but it should not fail as it does)